src/core/include/openvino/runtime/tensor_utils.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <filesystem>
+
+#include "openvino/core/partial_shape.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+
+/// \brief Save given tensor data into a file. File will contain only raw bytes of a tensor.data as it is allocated in
+///        memory. No element type nor shape nor other metadata are serialized. Strides are preserved.
+/// \param tensor Tensor which data will be serialized.
+/// \param file_name Path to the output file
+OPENVINO_API
+void save_tensor_data(const Tensor& tensor, const std::filesystem::path& file_name);
```

Comment:
how all these functions except `read_tensor_data` are related to mmap-ed file?

I would drop all non-mmap related functions.

Reply:
This is for feature completeness: to read something from a file, one should have a tool to write it to a file. Moreover, in the part that offloads existing constants to a file, we need to save tensor content.

Reply:
> don't work here?

Because it is more complex than `write_tensor_data(tensor, path`), it requires to know about .data and .get_byte_size, and BTW, it requires passing std::ios::binary and casting .data to char.

Reply:
> it requires to know about .data and .get_byte_size

User knows that he / she wants to mmap existing tensor, but does not know API of such tensors. Looks like not a real world scenario.

Reply:
Why not real? Our user wants to save memory while reading big tensors, user wants to save tensor content -- it is a higher-level API than accessing tensor elements. Even if he/she knows those things, why not hide this boilerplate (error-prone) code behind a convenient API?

Reply:
> code behind a convenient API?

put such helper function(s) to NNCF and use as convenient API

Reply:
@ilya-lavrenov numpy has save/load functions for np.ndarray, which has proven to be very popular and convenient API. IMHO: I believe that OpenVINO API will be more friendly if write/read functions will be present for ov.Tensor.

Reply:
I suppose the idea of this PR is to share mmap functionality of OpenVINO via mapping into Tensor, but not to create a lot of file utils for external tools.

Reply:
Let's keep is as is for 2025.1. This is enough for genai right now.
And discuss what do we need to add for nncf for the next release (as this is not in a scope right now)